src/main/java/org/broadinstitute/barclay/argparser/TaggedArgumentParser.java

```diff
@@ -0,0 +1,333 @@
+package org.broadinstitute.barclay.argparser;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.broadinstitute.barclay.utils.Utils;
+
+import java.util.*;
+
+/**
+ * The parser used by the {@link CommandLineArgumentParser} for handling tagged argument strings. Tagged arguments
+ * contain a tag name and optional attribute/value pairs, and are of the form:
+ *
+ *     --argument_name:logical_name argument_value
+ *
+ * or:
+ *
+ *     --argument_name:logical_name,key1=value1,key2=value2 argument_value
+ *
+ * The logical name is required; key/value pairs are optional. @Argument annotated fields that require tagged
+ * values must implement {@link TaggedArgument}, and the command line parser will require tag names for any such
+ * fields.
```

Comment:
Whoops, I think we misunderstood each other the other day when we were talking about whether tags should be required or not. It's not the case that arguments that implement `TaggedArgument` *must* have a tag. What I was saying was that if attributes are present, then a tag name must also be present, but it should be perfectly legal for a `TaggedArgument` field to have neither a tag nor attributes.